packages/flutter_web_plugins/lib/url_strategy.dart

```diff
@@ -5,4 +5,4 @@
 export 'src/navigation_common/url_strategy.dart';
 
 export 'src/navigation_non_web/url_strategy.dart'
-    if (dart.library.html) 'src/navigation/url_strategy.dart';
+    if (dart.library.ui_web) 'src/navigation/url_strategy.dart';
```

Comment:
IIRC this requires changes in the Dart compilers. Its not automatic from the libraries.json/yaml files

Reply:
Hmm, I thought otherwise but you might be right. Let me write a unit tests that tests that non-web can compile and gets the stubbed version

Reply:
Actually it appears that `if (dart.library.ui_web)` does actually work for me. Hmm.

Reply:
nice. I would check all compilers though :)

Reply:
I swear we added another constant here @sigmundch ?

Reply:
The list of libraries here depends precisely on the names in the `libraries.json` file used by the CFE when compiling the SDK. If you have a library that doesn't exist, this condition may simply be unconditionally false on all backends.

Seems you [recently](https://github.com/flutter/engine/pull/40608) added `dart:ui_web` as a public library, so this should be OK. In this case, it appears not only that the library exist, but it's precisely the one platform specific thing that you are using underneath in url_strategy.dart, so it seems reasonable to use this as your conditional import.

That said, there is a caveat in our internal build systems to consider. Our integration with bazel explicitly lists what can be used for configuration specific imports, but I can't recall at this moment how it does so (e.g. by listing specific libraries, or listing modes like "VM" vs "Web" vs "Flutter Web"). Bottomline - the BUILD rules are aware of itand the list of configurations is maintained by hand at the bazel/starlark level. Because it is not derived from the libraries.json, there is a small chance that a change will be needed whenever this rolls into g3.

Reply:
Thanks for the context! All the CI checks are passing, except the Google3 testing is giving issues, which is just as you've predicted.

Reply:
The alternative to the g3 issues could be to pick configurations that are already known there. For example `dart.library.js` or `dart.library.js_util`. Technically we don't want those if we are going to remove them in the wasm backend (which I think that's the plan). Ideally we would use `dart.library.js_interop` (the new library), but I'm guessing it probably has the same issue that `ui_web` has today.